accepted/2021/platform-dependencies/platform-dependencies.md

```diff
@@ -0,0 +1,433 @@
+
+## Stakeholders and Reviewers
+
+* ASP.NET Core
+    * [Kevin Pilch](https://github.com/Pilchie)
+* CoreCLR
+    * [Jeff Schwartz](https://github.com/jeffschwMSFT)
+* Acquisition & Deployment
+    * [Dan Leeaphon](https://github.com/dleeapho)
+* Docs
+    * [Andy De George](https://github.com/adegeo)
+* Libraries
+    * [Dan Moseley](https://github.com/danmoseley)
+* Mono
+    * [Marek Safar](https://github.com/marek-safar)
+* Release
+    * [Jamshed Damkewala](https://github.com/jamshedd)
+* SDK
+    * [Marc Paine](https://github.com/marcpopMSFT)
+
+## Design
+
+### Schema
+
+```csharp
+using System.Collections.Generic;
+
+namespace Microsoft.DotNet.Dependencies.Platform
+{
+    // Root of the model
+    public class PlatformDependencyModel
+    {
+        // Major.Minor.Patch version of .NET release the dependencies are associated with
+        public string DotnetReleaseVersion { get; set; }
+
+        // Set of available dependency usages that can be referenced by a platform dependency.
+        // A dependency usage is a descriptor of how a dependency is used by a component (e.g. default, diagnostics, localization).
+        // This maps the name of the name of the usage to its description.
+        public IDictionary<string, string> DependencyUsages { get; }
+        
+        // Set of top-level platforms whose dependencies are described
+        public IList<Platform> Platforms { get; }
+    }
+
+    // A supported platform
+    public class Platform
+    {
+        // Rid identifying the platform
+        public string Rid { get; set; }
+
+        // Child platforms (e.g. version-specific, arch-specific)
+        // Child platforms inherit the characteristics of their parents but can override them
+        public IList<Platform> Platforms { get; }
+
+        // Set of components and their dependencies specific to this platform
+        public IList<Component> Components { get; }
+    }
+
+    // A logical component within .NET that encapsulates a set of platform dependencies
+    public class Component
+    {
+        // Name of the component
+        public string Name { get; set; }
+
+        // Type of the component
+        public ComponentType Type { get; set; }
+
+        // Set of platform dependencies this component has
+        public IList<PlatformDependency> Dependencies { get; }
+    }
+
+    // Types of .NET components
+    public enum ComponentType
+    {
+        // A shared framework (e.g. Microsoft.NETCore.App, Microsoft.AspNetCore.App)
+        SharedFramework,
+
+        // A NuGet package (e.g. System.Drawing.Common)
```

Comment:
Since you are calling out System.Drawing.Common here, I can't help but wonder what that will look like. AFAIAA, `System.Drawing.Common` needs `libgdiplus`, but `libgdiplus` is neither a build-time nor a runtime dependency, unless you are explicitly using `System.Drawing.Common` at runtime. Does it fit in with the `default` type or does this need something else?

Reply:
@safern

Reply:
> unless you are explicitly using `System.Drawing.Common` at runtime

That's the key.  The dependency would specifically be tied to the `System.Drawing.Common` NuGet package.  So if you're referencing `System.Drawing.Common` at runtime, then you'll require `libgdiplus` for the canonical scenario.  In that case, the type would be `default`.

Here's a snippet of what that would look like:

```json
{
  "platforms":
  {
    "rid": "debian",
    "components": [
      {
        "name": "System.Drawing.Common",
        "type": "NuGetPackage",
        "platformDependencies": [
          {
            "name": "libgdiplus",
            "dependencyType": "LinuxPackage",
            "usage": "default"
          }
        ]
      }
    ]
  }
}
```